lighthouse-core/report/html/renderer/snippet-list-renderer.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+/* globals self, Util */
+
+/** @typedef {import('./details-renderer')} DetailsRenderer */
+
+/**
+ * @param {LH.Audit.DetailsRendererSnippetLine[]} lines
+ * @param {number} lineNumber
+ */
+function getLineAndPreviousLine(lines, lineNumber) {
+  return {
+    line: lines.find(l => l.number === lineNumber),
+    previousLine: lines.find(l => l.number === lineNumber - 1),
+  };
+}
+
+class SnippetListRenderer {
```

Comment:
I feel like we need an overview for this type/renderer somewhere, though I'm not sure exactly where. Basically the only way to understand how it works right now is to read this file :)

In this file? On the details type in `audit.d.ts`?

Reply:
Added some comments in `audit-details.d.ts`.